service_management/azure/lib/azure/client_services.rb

```diff
@@ -92,7 +92,8 @@ def base_management(options = {})
     private
 
     def default_client(opts)
-      {client: Azure.client(opts || {})}
+      !opts.empty? ? {client: Azure.client(opts || {})} : {client: self}
```

Comment:
As you mentioned offline that we can rewrite this as ` !opts.empty? ? {client: Azure.client(opts)} : {client: self}`

@veronicagg Other things looks great.